src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentKind.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Indicates kind of a <see cref="TextDocument"/>
+    /// </summary>
+    /// <remarks>CONSIDER: Make this type public</remarks>
+    internal enum TextDocumentKind
+    {
```

Comment:
with enum. adding new kind breaking change?

Reply:
I am not sure - I have stayed defensive here and kept it internal only for now.